misk/src/main/kotlin/misk/web/interceptors/MetricsInterceptor.kt

```diff
@@ -12,26 +14,31 @@ import javax.inject.Singleton
 
 internal class MetricsInterceptor internal constructor(
   private val actionName: String,
-  private val requestDuration: Histogram
+  private val requestDuration: Histogram,
+  private val caller: ActionScoped<MiskCaller?>
 ) : NetworkInterceptor {
   override fun intercept(chain: NetworkChain): Response<*> {
     val (elapsedTime, result) = timed { chain.proceed(chain.request) }
 
     val elapsedTimeMillis = elapsedTime.toMillis().toDouble()
-    requestDuration.labels(actionName, "all").observe(elapsedTimeMillis)
-    requestDuration.labels(actionName, "${result.statusCode / 100}xx").observe(elapsedTimeMillis)
-    requestDuration.labels(actionName, "${result.statusCode}").observe(elapsedTimeMillis)
+    val callingPrincipal = caller.get()?.principal ?: "unknown"
```

Comment:
Is this a cardinality explosion?

Reply:
It's the same cardinality we have today (and it's kindof critical to identifying poorly behaved clients). We can aggregate all clients into a single timeseries via recording rules on the Prometheus side.